nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java

```diff
@@ -386,17 +398,26 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             }
             this.setMessageHeader("X-Mailer", context.getProperty(HEADER_XMAILER).evaluateAttributeExpressions(flowFile).getValue(), message);
             message.setSubject(context.getProperty(SUBJECT).evaluateAttributeExpressions(flowFile).getValue());
+            message.setSentDate(new Date());
 
             final String messageText = getMessage(flowFile, context, session);
-
             final String contentType = context.getProperty(CONTENT_TYPE).evaluateAttributeExpressions(flowFile).getValue();
-            message.setContent(messageText, contentType);
-            message.setSentDate(new Date());
+            final Charset charset = getCharset(context);
+            final String charsetName = MimeUtility.mimeCharset(charset.name());
+            final DataHandler messageDataHandler = new DataHandler(
+                    new ByteArrayDataSource(
+                            Base64.encodeBase64(messageText.getBytes(charset)),
+                            contentType + String.format("; charset=\"%s\"", charsetName)
+                    )
+            );
+
+            final MimeMultipart multipart = new MimeMultipart();
+            final MimeBodyPart mimeText = new PreencodedMimeBodyPart("base64");
+            mimeText.setDataHandler(messageDataHandler);
+            mimeText.setHeader("Content-Transfer-Encoding", "base64");
+            multipart.addBodyPart(mimeText);
```

Comment:
This approach changes the message format to always use multipart formatting, which does not seem the like the ideal solution.